### PR TITLE
Remove ajv-formats-draft2019

### PR DESCRIPTION
**Pull request by pasieronen**
### Description

Replace unmaintained dependency ajv-formats-draft2019 with our own implementation. 
Continues work started by grease-work-23 in #1226. See also discussion in #1223 and 
https://github.com/CycloneDX/cyclonedx-webpack-plugin/issues/1348

The only piece we still used from ajv-formats-draft2019 is "idn-email". I tried to keep the code simple, and re-use ajv-formats (which we anyway depend on) as much as possible.

It might be possible to implement "iri-reference" in similar way (convert to URI, and use "uri-reference" from ajv-fomats), but I'm not sure how useful that would be: almost any string is a valid "uri-reference", so it wouldn't probably catch any problems anyway.

Resolves or fixes issue: #1226 
supersedes #1226

### AI Tool Disclosure

- [X] My contribution does not include any AI-generated content

### Affirmation

- [X] My code follows the [CONTRIBUTING.md](https://github.com/CycloneDX/cyclonedx-javascript-library/blob/main/CONTRIBUTING.md) guidelines

### CODE REVIEW

**Review by jkowalleck — CHANGES_REQUESTED** [review]
Thank you so much for taking care of this topic. 

see my remarks below

File: src/_optPlug.node/__jsonValidators/ajv.ts [hunk 1]
```diff
@@ -50,11 +48,19 @@ export default (async function (schemaPath: string, schemaMap: Record<string, st
...
   const ajv = new Ajv({ ...ajvOptions, schemas })
   addFormats(ajv)
-  /* eslint-disable-next-line @typescript-eslint/no-unsafe-call -- intended */
-  addFormats2019(ajv, { formats: ['idn-email'] })
+
   // there is just no working implementation for format "iri-reference": see https://github.com/luzlab/ajv-formats-draft2019/issues/22
   ajv.addFormat('iri-reference', true)
 
+  // add idn-email format (was previously provided by ajv-formats-draft2019)
+  const emailValidator = ajv.compile({type: 'string', format: 'email'})
```
[the remark this answers is not in the document]
Comment (pasieronen) [reply src/_optPlug.node/__jsonValidators/ajv.ts]: Sure! Especially the empty string is a good idea, added couple more at the same time (amended the same commit).

I also noticed that not even ajv-formats and ajv-formats-draft2019 agree on every input. For example, `"John Doe"@example.com` is valid according to the latter but not the former. That said, I don't expect this to be a problem (and we probably don't want too many tests for rare edge cases, since ajv-formats itself may also change over time).

**Review by CAOShurong — CHANGES_REQUESTED** [review]
The dependency removal and the existing test suite pass on exact head `1fece39dd1b9bd367a82daaa0c28ccc794e39454`, but this head changes an accepted `idn-email` case that falls directly within the unresolved edge-case discussion.

Using the real CycloneDX 1.7 validator, I reproduced this boundary on the default-branch base `c3d8535` and the PR head:

- `"John Doe"@example.com`: accepted by the current default branch, rejected by this PR;
- `用户@例子.广告`: accepted by both;
- `johndoe`: rejected by both.

The dependency being replaced also explicitly tests `"John Doe"@example.com` as valid (`ajv-formats-draft2019/index.test.js`), and the discrepancy is already acknowledged in the current unresolved thread. This is therefore a concrete compatibility change, not a hypothetical difference between validators.

Please either preserve the existing quoted-local-part behavior and add it as a regression test, or explicitly document and justify the compatibility break before merging. Validation on this exact head included 18 focused JSON-validator tests, the full 4,180-test Node suite, TypeScript, ESLint, and Knip; all passed, so the request is limited to the uncovered compatibility boundary above.

Review assisted by OpenAI Codex; I independently checked the exact head and reproduced the three validator cases above.
